.github/workflows/test.yaml

```diff
@@ -284,28 +283,30 @@ jobs:
         with:
           go-version: 1.22.x
         id: go
-      - name: Check out code
+      - name: Checkout code at git ref
         uses: actions/checkout@v4
-      - name: Run make verify-manifests
-        run: |
-          make verify-manifests
+      - name: Check bundles and manifests
+        run: make verify-prepare-release
 
-  verify-bundle:
-    name: Verify bundle
-    if: needs.pre-job.outputs.should_skip != 'true'
+  verify-manifests:
```

Comment:
I think it is better to have N verify jobs. Two reasons:
* the UX is better in Github. You know very quickly what failed
* Each tests is run in isolation: one test cannot interfere the other test. For instance, `make bundle` could be changing files from helm charts and the helm charts test could be failing due to bundle test.  Test isolation is always a desired property.

Reply:
I am not sure I fully agree with the each test running in isolation. There is a strong interaction between the bundles and helm charts when it comes to releasing. Would see this test more of an integration test over a unit test. When the `make prepare-release` would resolve both issue I don't see the need to have them split.

You might know very quickly what failed, but does that tell you as quickly how to fix it. I also think the make target in the test should not be used, but instead use `make verify-prepare-release`.

Reply:
The source of truth is in golang code and `/config` plus few other places like Makefile. Those are verification steps to make sure the source of truth is consistent with the shipped manifests (bundle or helm charts). So, IMO, independent and isolated tests are good to have. I can live without it as well.

Reply:
In the future, the Makefile will no longer be a source of truth.. it will be replaced by release.yaml. Because now there are two ways to generate the manifests and eventually we need to merge into one.

Reply:
Agreed eventually it needs to merge into one. The next step on the road I see is how the nightly test pipeline generate their builds.

Reply:
@eguzki I could add a commit splitting the verification step in 3, as it was before